state/txindex/kv/kv_test.go

```diff
@@ -692,8 +692,18 @@ func TestTxSearchMultipleTxs(t *testing.T) {
 
 	results, err := indexer.Search(ctx, query.MustParse("account.number >= 1"))
 	assert.NoError(t, err)
-
 	require.Len(t, results, 3)
+
+	// since two txs were added at height 1 and 2, we should have two unique transactions
```

Comment:
I added more tests to make sure that transaction searches return unique results and accurate number of transactions included at a specific block height. While this logic is already covered in the unit tests for this file, the existing implementation seems to work correctly in the local repository even without applying the fix. 

It was manually tested by me and the NodesGuru team. https://github.com/celestiaorg/celestia-core/issues/1526#issuecomment-2479340821